Avoid an implicit int and an implicit declaration of exit.  This
prevents build failures with future compilers.

diff --git a/configure.in b/configure.in
index 1a0ad1be03f9f9fe..011c1d919885536a 100644
--- a/configure.in
+++ b/configure.in
@@ -366,11 +366,11 @@ AC_TRY_RUN([
 
 struct linger li;
 
-main ()
+int main (void)
 {
     li.l_onoff = 1;
     li.l_linger = 120;
-    exit (0);
+    return 0;
 }
 ],[
 AC_DEFINE(HAVE_STRUCT_LINGER, 1, [Define to 1 if struct linger is available])
